src/components/orator/js/Orator.js

```diff
@@ -0,0 +1,963 @@
+            utteranceOnEnd: null,
+            utteranceOnBoundary: null,
+            utteranceOnError: null,
+            utteranceOnMark: null,
+            utteranceOnPause: null,
+            utteranceOnResume: null,
+            utteranceOnStart: null
+        },
+        utteranceEventMap: {
+            onboundary: "utteranceOnBoundary",
+            onend: "utteranceOnEnd",
+            onerror: "utteranceOnError",
+            onmark:"utteranceOnMark",
+            onpause: "utteranceOnPause",
+            onresume: "utteranceOnResume",
+            onstart: "utteranceOnStart"
+        },
+        model: {
+            tts: {
+                paused: false,
+                speaking: false,
+                enabled: true
+            },
+            parseQueuelength: 0,
+            parseIndex: null,
+            ttsBoundary: null
+        },
+        modelRelay: [{
+            target: "parseIndex",
+            backward: "never",
+            namespace: "getClosestIndex",
+            singleTransform: {
+                type: "fluid.transforms.free",
+                func: "fluid.orator.domReader.getClosestIndex",
+                args: ["{that}", "{that}.model.ttsBoundary"]
+            }
+        }],
+        members: {
+            parseQueue: [],
+            range: {
+                expander: {
+                    this: "document",
+                    method: "createRange"
+                }
+            }
+        },
+        components: {
+            parser: {
+                type: "fluid.orator.domReader.parser"
+            }
+        },
+        invokers: {
+            readFromDOM: {
+                funcName: "fluid.orator.domReader.readFromDOM",
+                args: ["{that}", "{that}.container"]
+            },
+            removeHighlight: {
+                funcName: "fluid.orator.domReader.unWrap",
+                args: ["{that}.dom.highlight"]
+            },
+            setParseQueue: {
+                funcName: "fluid.orator.domReader.setParseQueue",
+                args: ["{that}", "{arguments}.0"]
+            },
+            resetParseQueue: {
+                funcName: "fluid.orator.domReader.setParseQueue",
+                args: ["{that}"]
+            },
+            highlight: {
+                funcName: "fluid.orator.domReader.highlight",
+                args: ["{that}", "{arguments}.0"]
+            },
+            play: {
+                funcName: "fluid.orator.domReader.speak",
+                args: ["{that}", "{fluid.textToSpeech}.resume"]
+            },
+            pause: {
+                funcName: "fluid.orator.domReader.pause",
+                args: ["{that}", "{fluid.textToSpeech}.pause"]
+            },
+            queueSpeech: {
+                funcName: "fluid.orator.domReader.queueSpeech",
+                args: ["{that}", "{arguments}.0", true, "{arguments}.1"]
+            }
+        },
+        modelListeners: {
+            "parseIndex": {
```

Comment:
As you mentioned in our call, this might also be invalidated by a change in parseQueueLength that does not at the same time change parseIndex. You can use the "path" member to listen to both at once: https://docs.fluidproject.org/infusion/development/ChangeApplierAPI.html#the-path-entry-in-a-model-listener-declaration